pkg/export/setup/setup.go

```diff
@@ -239,6 +249,43 @@ func ExtraArgs() ([]string, error) {
 	return shlex.Split(os.Getenv(ExtraArgsEnvvar))
 }
 
+type MetadataOpts struct {
```

Comment:
question: Why separate the MetadataOpts config?

And either way, if want to do that, it would be much easier to review in a dedicated PR with a justification provided as there's a lot going on here already :)

Reply:
Addressing in https://github.com/GoogleCloudPlatform/prometheus-engine/pull/1001